src/tribol/interface/mfem_tribol.cpp

```diff
@@ -119,6 +118,133 @@ void setMfemLORFactor( integer cs_id,
    coupling_scheme->getMfemMeshData()->SetLORFactor(lor_factor);
 }
 
+void setMfemKinematicConstantPenalty( integer cs_id, 
+                                      real mesh1_penalty,
+                                      real mesh2_penalty )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemData(),
+      "Coupling scheme does not contain MFEM data. "
+      "Create the coupling scheme using registerMfemCouplingScheme() to set the penalty."
+   );
+   setPenaltyOptions(cs_id, KINEMATIC, KINEMATIC_CONSTANT);
+   coupling_scheme->getMfemMeshData()->ClearAllPenaltyData();
+   coupling_scheme->getMfemMeshData()->SetMesh1Penalty(mesh1_penalty);
+   coupling_scheme->getMfemMeshData()->SetMesh2Penalty(mesh2_penalty);
+}
+
+void setMfemKinematicElementPenalty( integer cs_id, 
+                                     mfem::Coefficient& modulus_coefficient )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemData(),
+      "Coupling scheme does not contain MFEM data. "
+      "Create the coupling scheme using registerMfemCouplingScheme() to set the penalty."
+   );
+   setPenaltyOptions(cs_id, KINEMATIC, KINEMATIC_ELEMENT);
+   coupling_scheme->getMfemMeshData()->ClearAllPenaltyData();
+   coupling_scheme->getMfemMeshData()->ComputeElementThicknesses();
+   coupling_scheme->getMfemMeshData()->SetMaterialModulus(modulus_coefficient);
+}
+
+void setMfemRateConstantPenalty( integer cs_id, 
+                                 real mesh1_penalty, 
+                                 real mesh2_penalty )
+{
+   auto coupling_scheme = CouplingSchemeManager::getInstance().getCoupling(cs_id);
+   SLIC_ERROR_ROOT_IF(
+      !coupling_scheme->hasMfemData(),
+      "Coupling scheme does not contain MFEM data. "
+      "Create the coupling scheme using registerMfemCouplingScheme() to set the penalty."
+   );
+   auto penalty_opts = coupling_scheme->getEnforcementOptions().penalty_options;
+   SLIC_ERROR_ROOT_IF(
+      !penalty_opts.kinematic_calc_set,
+      "No kinematic enforcement method set. Call setMfemKinematicConstantPenalty() or "
+      "setMfemKinematicElementPenalty() first."
+   );
+   setPenaltyOptions(
+      cs_id, penalty_opts.constraint_type, penalty_opts.kinematic_calculation, RATE_CONSTANT);
```

Comment:
penalty_opts.constraint_type should be updated/changed to KINEMATIC_AND_RATE (or whatever the enum is).